packages/core/src/external/commonwell/document/document-downloader-local.ts

```diff
@@ -98,28 +99,36 @@ export class DocumentDownloaderLocal extends DocumentDownloader {
       return { ...downloadResult };
     }
 
+    const old_extension = path.extname(fileInfo.name);
     const [detectedFileType, detectedExtension] = detectFileType(downloadedDocument);
 
-    console.log(
-      `Updating content type in S3 ${fileInfo.name} from previous mimeType: ${document.mimeType} to detected mimeType ${detectedFileType} and ${detectedExtension}`
-    );
-
-    const newKey = await this.s3Utils.updateContentTypeInS3(
-      downloadResult.bucket,
-      downloadResult.key,
-      detectedFileType,
-      detectedExtension
-    );
+    console.log("old extension: ", old_extension);
+    // If the file type has changed
+    if (detectedFileType !== document.mimeType && old_extension !== detectedExtension) {
```

Comment:
check that the mime type and extension have changed. This stops us from deleting files that havent changed

Reply:
NABD, but it's easier to read the code if we do the logic the other way around, then we can do something like this:

```ts
if (detectedFileType === document.mimeType || old_extension === detectedExtension) {
  return downloadResult;
}
...
```

Then the whole code is not inside a conditional.